File: app/src/main/java/jp/bizloco/kotlinexample/RecyclerViewAdapter.java
package jp.bizloco.kotlinexample;

import android.annotation.SuppressLint;
import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;

import java.util.List;

/**
 * Created by PC on 11/23/2016.
 */

public class RecyclerViewAdapter extends BaseAdapter {

    /**
     * click item quote interface
     */
    public interface ItemClickListener {
        void onClickItem(int position);
    }

    private ItemClickListener mListener;
    private List<String> list;
    private Context mContext;

    public RecyclerViewAdapter(Context context, List<String> array, ItemClickListener listener) {
        super(context);
        this.mContext = context;
        this.list = array;
        this.mListener = listener;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_recycler_view, parent, false);
        return new ImageHolder(view);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        bindImageHolder((ImageHolder) holder, position);
    }

    @SuppressLint("CheckResult")
    private void bindImageHolder(ImageHolder imageHolder, int position) {
        if (list.get(position) !=null){
            imageHolder.mImgItem.layout(0, 0, 0, 0);
            RequestOptions requestOptions = new RequestOptions();
            requestOptions.placeholder(R.drawable.loading_photo);
            Glide.with(mContext)
                    .setDefaultRequestOptions(requestOptions)
                    .load(list.get(position).trim())
                    .into(imageHolder.mImgItem);
        }
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    private class ImageHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        private ImageView mImgItem;

        private ImageHolder(View itemView) {
            super(itemView);
            mImgItem = itemView.findViewById(R.id.mImgItem);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            if (mListener != null) {
                mListener.onClickItem(getLayoutPosition());
            }
        }
    }
}

File: app/src/main/java/jp/bizloco/kotlinexample/RecyclerViewAdapter.kt
package jp.bizloco.kotlinexample

import android.annotation.SuppressLint
import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import kotlinx.android.synthetic.main.item_recycler_view.view.*

/**
 * Created by PC on 11/23/2016.
 */

class RecyclerViewAdapter(var mContext: Context, var mList: ArrayList<String>, var mListener: ItemClickListener) : BaseAdapter<RecyclerViewAdapter.ImageHolder>(mContext) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ImageHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_recycler_view, parent, false)
        return ImageHolder(view)
    }

    override fun onBindViewHolder(holder: ImageHolder, position: Int) {
        bindImageHolder(holder as ImageHolder, position)
    }

    @SuppressLint("CheckResult")
    private fun bindImageHolder(imageHolder: ImageHolder, position: Int) {
        imageHolder.mImgItem.layout(0, 0, 0, 0)
        val requestOptions = RequestOptions()
        requestOptions.placeholder(R.drawable.loading_photo)
        Glide.with(mContext)
                .setDefaultRequestOptions(requestOptions)
                .load(mList.get(position).trim())
                .into(imageHolder.mImgItem)

        imageHolder.mImgItem.setOnClickListener { mListener.onClickItem(position) }
    }

    override fun getItemCount(): Int {
        return mList.size
    }

    class ImageHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val mImgItem = itemView.mImgItem
    }
}
